### PR TITLE
Make events parameter required for ListEvents

### PR DESCRIPTION
**Pull request by mattgd**
## Description
Make events parameter required for ListEvents.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[x] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.
https://github.com/workos/workos/pull/26139

### CODE REVIEW

**Review by PaulAsjes — APPROVED** [review]

File: pkg/events/client.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ type Event struct {
 // ListEventsOpts contains the options to request provisioned Events.
 type ListEventsOpts struct {
 	// Filter to only return Events of particular types.
-	Events []string `url:"events,omitempty"`
+	Events []string `url:"events"`
```
Comment (mattgd) [inline pkg/events/client.go:92]: @PaulAsjes While this change is breaking in the API (mitigated by feature flag for affected customers), I don't consider this is a breaking change in the Go SDK, so a minor bump seems appropriate. Wanted to double check to see if you had an opinion though.

Comment (PaulAsjes) [reply pkg/events/client.go:92]: Agreed, minor bump seems fine!

File: internal/workos/workos.go [hunk 2]
```diff
@@ -2,5 +2,5 @@ package workos
 
 const (
 	// Version represents the SDK version number.
-	Version = "v4.5.0"
+	Version = "v4.6.0"
```
Comment (mthadley) [inline internal/workos/workos.go:5]: Can we bump the version in a separate PR? Like we've done [here](https://github.com/workos/workos-go/pull/321) and in earlier releases.

There are other unreleased changes on `main`, so it makes the commit history easier to view in regards to what was included in each version bump.

Comment (mattgd) [reply internal/workos/workos.go]: Yep for sure, will split.

**Review by mthadley — APPROVED** [review]

**Pull request merged**